libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -961,11 +964,13 @@ void NavEKF3_core::calcOutputStates()
 }
 
 /*
- * Calculate the predicted state covariance matrix using algebraic equations generated with Matlab symbolic toolbox.
- * The script file used to generate these and other equations in this filter can be found here:
- * https://github.com/PX4/ecl/blob/master/matlab/scripts/Inertial%20Nav%20EKF/GenerateNavFilterEquations.m
+ * Calculate the predicted state covariance matrix using algebraic equations generated using SymPy
+ * See AP_NavEKF3/derivation/main.py for derivation
+ * Output for change reference: AP_NavEKF3/derivation/generated/covariance_generated.cpp
+ * Argument rotVarVecPtr is pointer to a vector defining the earth frame uncertainty variance of the quaternion states
+ * used to perform a reset of the quaternion state covariances only. Set to null for normal operation.
 */
-void NavEKF3_core::CovariancePrediction()
+void NavEKF3_core::CovariancePrediction(Vector3f *rotVarVecPtr)
```

Comment:
We very rarely pass around pointers because of the danger of a memory access violation.  It's not a blocker but I hope we have a check high in the function to catch a nullptr.  I'm thinking of future changes to this function where the developer doesn't realise that a nullptr could be passed in.

Reply:
There is a check.